src/fpm_meta.f90

```diff
@@ -1005,9 +1015,8 @@ subroutine init_mpi_from_wrappers(this,compiler,mpilib,fort_wrapper,c_wrapper,cx
     end if
 
     !> Add default run command, if present
-    this%run_command = mpi_wrapper_query(mpilib,fort_wrapper,'runner',verbose,error)
-    if (allocated(error)) return
-    this%has_run_command = len_trim(this%run_command)>0
+    this%run_command = mpi_wrapper_query(mpilib,fort_wrapper,'runner',verbose,runner_error)
+    this%has_run_command = (len_trim(this%run_command)>0) .and. .not.allocated(runner_error)
```

Comment:
These lines before the change look pretty good to me. What speaks against them? We return early if there's an error and therefore do not need to care about the value of `this%has_run_command`.

Reply:
These were changed to account for your request @minhqdao. If you think the previous version was better, please revert, thank you.

Reply:
No, the question was what speaks against the lines 1008-1010 (before this PR)?

Reply:
Confirm my previous answer